lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -248,6 +252,30 @@ defmodule KinesisClient.Stream.Shard.Producer do
             message:
               "Checkpoint failed for #{state.app_name}-#{state.shard_id}: this consumer is not the lease owner"
 
+        {:error, {"ProvisionedThroughputExceededException", _}} ->
+          if attempt > @max_update_checkpoint_retries do
+            :telemetry.execute(
+              [:kinesis_client, :shard_producer, :checkpoint_error],
+              %{reason: :provisioned_throughput_exceeded, fatal: true},
+              telemetry_meta(state)
+            )
```

Comment:
One last nit: could we add this Telemetry event also to the fallback "unknown" clause below?

Reply:
We're already tracking the `span` which emits an exception with reason already. Sentry captures the full detail of the information as well, so I think we're already good on that front?